### PR TITLE
Fix non-compile error

### PR DESCRIPTION
**Pull request by conradjones**
Added CMakeLists.txt for compilation with CMake and any IDE that supports CMake.

Removed const qualifier from function parameter where function modifiers parameter.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -46,7 +46,7 @@ TEST_CASE("Basic usage")
 
 using NameRef = fluent::NamedType<std::string&, struct NameRefParameter>;
 
-void changeValue(const NameRef name)
+void changeValue(NameRef name)
```
Comment (knatten) [inline main.cpp]: Nice that you fixed this compilation error!

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+cmake_minimum_required(VERSION 3.9)
+project(NamedType)
```
Comment (knatten) [inline CMakeLists.txt]: Maybe change this to `project(NamedType CXX)` to indicate that it's a C++ project.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+cmake_minimum_required(VERSION 3.9)
```
Comment (knatten) [inline CMakeLists.txt]: Any reason to require a cmake version this recent? Could be good to allow users on older cmake versions to use this as well.

**Comment by knatten** [comment]
This fixes #23

**Comment by knatten** [comment]
Could be an idea to split out the compilation error fix as a separate PR which will be a no-brainer to accept?

**Comment by travnick** [comment]
see #37

File: CMakeLists.txt (hunk 3 above)
Comment (joboccara) [reply CMakeLists.txt:1]: how about using 3.12? This is the version I use in the pipes library

**Comment by joboccara** [comment]
This should be covered by PR #24

**Pull request closed without merging**